src/LineSegment.cpp

```diff
@@ -173,6 +173,9 @@ Vector2 LineSegment::project(const Vector2 &point) const {
 }
 
 bool LineSegment::isOnLine(const Vector2 &point) const {
+    if (isPoint()) {
+        return start == point;
+    }
```

Comment:
This is dangerous. LineSegments are in constrast to Vector2's not treated with the precision margin because errors can scale up infinitely due to extrapolation/scaling along the line. Implying that small line segments of size<10^-5 are points might be quite problematic.

Reply:
As in, this could for example actually return false on end==point but true on start==point and vice versa.

Reply:
I will fix this...